From 3d7d5c10c806ee24f8e35014e41e203002609a3f Mon Sep 17 00:00:00 2001
From: Andreas Schwab <schwab@suse.de>
Date: Thu, 8 Sep 2022 10:25:21 +0200
Subject: [PATCH] errlist: add missing entry for EDEADLOCK (bug 29545)

Some architectures (mips, powerpc and sparc) define separate values for
EDEADLOCK and EDEADLK.  Readd the errlist entry for EDEADLOCK for those
configurations.  Also use the dependency files from generating the
auxiliary errlist and siglist files.
---
 stdio-common/Makefile | 8 ++++++++
 sysdeps/gnu/errlist.h | 3 +++
 2 files changed, 11 insertions(+)

diff --git a/stdio-common/Makefile b/stdio-common/Makefile
index e11aaedd96..9c98c02884 100644
--- a/stdio-common/Makefile
+++ b/stdio-common/Makefile
@@ -267,6 +267,10 @@ $(objpfx)errlist-data-aux.S: errlist-data-gen.c
 	$(make-target-directory)
 	$(compile-command.c) $(pie-default) $(no-stack-protector) -S
 
+ifndef no_deps
+-include $(objpfx)errlist-data-aux.S.d $(objpfx)errlist-data-aux-shared.S.d
+endif
+
 $(objpfx)errlist-data.os: $(objpfx)errlist-data-aux-shared.S
 $(objpfx)errlist-data.o: $(objpfx)errlist-data-aux.S
 
@@ -278,6 +282,10 @@ $(objpfx)siglist-aux.S: siglist-gen.c
 	$(make-target-directory)
 	$(compile-command.c) $(pie-default) $(no-stack-protector) -S
 
+ifndef no_deps
+-include $(objpfx)siglist-aux.S.d $(objpfx)siglist-aux-shared.S.d
+endif
+
 $(objpfx)siglist.os: $(objpfx)siglist-aux-shared.S
 $(objpfx)siglist.o: $(objpfx)siglist-aux.S
 
diff --git a/sysdeps/gnu/errlist.h b/sysdeps/gnu/errlist.h
index 6329e5f393..e8416448da 100644
--- a/sysdeps/gnu/errlist.h
+++ b/sysdeps/gnu/errlist.h
@@ -593,6 +593,9 @@ _S(EBADRQC, N_("Invalid request code"))
 #ifdef EBADSLT
 _S(EBADSLT, N_("Invalid slot"))
 #endif
+#if defined EDEADLOCK && EDEADLOCK != EDEADLK
+_S (EDEADLOCK, N_ ("File locking deadlock error"))
+#endif
 #ifdef EBFONT
 _S(EBFONT, N_("Bad font file format"))
 #endif
-- 
2.37.3

